### DISCUSSION TITLE
valdiate_object: schema keys must be a subset of data keys

### DISCUSSION THREAD

**Discussion by H4rryK4ne** [bhch/django-jsonform#178] [Ideas]
Hi @bhch 

thanks for this awesome library. I have a question to the implementation detail. 

In [validators.validate_object(...)](https://github.com/bhch/django-jsonform/blob/d815696b380d5ffd35a735645cdac8adff1af450/django_jsonform/validators.py#L212) you require, that all keys of the schema need to be present, even if they are not required. 

Is there a special reason for this limitation? 

I identified one possible "reason", in the loop starting at [line L239](https://github.com/bhch/django-jsonform/blob/d815696b380d5ffd35a735645cdac8adff1af450/django_jsonform/validators.py#L239). But this could be handled in a different way:

Instead of 
```
for key in data:
```
One could do something like:
```
for key in {*data.keys(), *schema_keys}
```
If a `key` is not existing in `data` it could be handled like a `key` with `None` value.

Would you accept a PR for this change?

Best regards
Harry

**Comment by bhch** [comment]
Reasoning is to match Django's behaviour. More detailed answer: https://github.com/bhch/django-jsonform/issues/109#issuecomment-1592743979.

Can you give me some examples why you need this behaviour changed? What issues are you facing due to this?
